apps/oauth2/lib/Controller/LoginRedirectorController.php

```diff
@@ -76,12 +81,28 @@ public function authorize($client_id,
 
 		$this->session->set('oauth.state', $state);
 
-		$targetUrl = $this->urlGenerator->linkToRouteAbsolute(
-			'core.ClientFlowLogin.showAuthPickerPage',
-			[
-				'clientIdentifier' => $client->getClientIdentifier(),
-			]
-		);
+		if (in_array($client->getName(), $this->appConfig->getValueArray('oauth2', 'autoGrantApplications', []))) {
+			/* See ClientFlowLoginController::showAuthPickerPage */
```

Comment:
```suggestion
			/** @see ClientFlowLoginController::showAuthPickerPage */
```
To make it clickable?